### DISCUSSION TITLE
HOW TO ENABLE CSV FILE UPLOAD ,showing ERROR: No database is allowed for your csv upload

### DISCUSSION THREAD

**Discussion by anoop-pm** [apache/superset#14143] [Q&A / Help] [answered]

**Comment by suddjian** [comment]
1. Go to Databases page
2. Open the edit modal for the database you want to upload to
3. Go to the "Extra" tab
4. Check "Allow data upload"
5. Follow directions on the page to set the schemas that should be available for upload

**Reply by bjshijh** [reply]
i can not see "Extra" tab in version 1.3.2. what to do?

**Reply by srinify** [reply]
Hi @bjshijh you will see "Extra" in the last phase (depending on which db) of the 'connect a database' wizard. 

Can you share a screenshot of what you're seeing?

**Reply by arminus** [reply]
Chiming in here b/c I have the same problem, don't see an extras tab as shown in the [docs](https://superset.apache.org/docs/creating-charts-dashboards/exploring-data#loading-csv-data)

Cloned form git last week, master branch, here's how the edit dialog for the examples DB looks like:

![2021-11-15_123549](https://user-images.githubusercontent.com/683680/141775679-d718c707-01c9-418c-b8c9-802cdaea1de8.png)

There's nothing to the right of the Advanced tab.

**Reply by arminus** [reply]
Seems this is now in Advanced -> Security -> Allow data upload

**Comment by srinify** [comment]
For future reference, the instructions @suddjian pointed out, are in the Superset docs: https://superset.apache.org/docs/creating-charts-dashboards/exploring-data#loading-csv-data

**Comment by anoop-pm** [comment]
thanku so much, Also i solved my problem

**Comment by sroy-eq** [comment, marked as answer]
This is what worked for me 
Open edit of the database where you want to upload the CSV -> Advanced -> Security -> Select Allow Data Upload (towards the 
bottom). 

Then use the `Upload file to database` menu to upload the file. 

Hope it helps.

**Reply by JakobMiksch** [reply]
this setting is quite hidden

**Comment by tomtom215** [comment]
Can anyone assist? This setting is no longer in the CSV -> Advanced -> Security section. The docs also are outdated.

**Reply by tomtom215** [reply]
Something I noticed, I was using ClickHouse as a DB and the CSV option was never available, but it was with Postgres

**Comment by anoop-pm** [comment]
I do that. But the data table name must be same

**Comment by srinify** [comment]
Just a heads up -- in general, the upload CSV file to database is very much oriented around workarounds / temporary unblocking instead of as a full-on, high quality CSV import tool for end-users. It has a LOT missing and it's helpful to think about it more as an "escape valve"!

**Comment by Juliodonadello** [comment]
Someone found how to upload a .csv to the "examples" database?
I am trying checking the Secutiy-->Allow file uploads to database  and writing the main schema as the one allowed, but i get this error: "An error occurred while fetching databases: {"sqlalchemy_uri":["SQLiteDialect_pysqlite cannot be used as a data source for security reasons."]}"
I have to create a new database connection in order to upload a .csv?

**Reply by leandroramos** [reply]
I'm trying it too, got the same error.

**Reply by pmalonis** [reply]
I fixed this by setting "PREVENT_UNSAFE_DB_CONNECTIONS = False" my superset_config.py file. The documentation on this point is really terrible, especially considering it is the first step in the tutorial after installation.

**Comment by KansaiTraining** [comment]
I wonder why they keep changing completely the menus. Can someone assist with where should I set to allow uploading csv? My version is 1.5.1 and my screenshot is this
![Screenshot from 2024-06-28 13-13-04](https://github.com/apache/superset/assets/18393739/0497d021-3708-400f-88de-638b75d5f418)

**Reply by JavierMorales2** [reply]
> @KansaiTraining you need to create a database connection first

**Comment by ygbingo** [comment]
When you create database, you could check this can upload csv.
```Allow Csv Upload```
